common/extended_window.hpp

```diff
@@ -6,42 +6,44 @@
 #include <GL/glx.h>
 #include <GL/glu.h>
 #include "phonebook.hpp"
+#include "global_module_defs.hpp"
 
 //GLX context magics
 #define GLX_CONTEXT_MAJOR_VERSION_ARB       0x2091
 #define GLX_CONTEXT_MINOR_VERSION_ARB       0x2092
 typedef GLXContext (*glXCreateContextAttribsARBProc)(Display*, GLXFBConfig, GLXContext, Bool, const int*);
 
-namespace ILLIXR{
+namespace ILLIXR {
     class xlib_gl_extended_window : public phonebook::service {
     public:
-        int                     width;
-        int                     height;
-        Display                 *dpy;
-        Window                  win;
-        GLXContext              glc;
+        int        width;
+        int        height;
+        Display    *dpy;
+        Window     win;
+        GLXContext glc;
         xlib_gl_extended_window(int _width, int _height, GLXContext _shared_gl_context){
             width = _width;
             height = _height;
 
 #ifndef NDEBUG
-            printf("Opening display\n");
+            std::cout << "Opening display" << std::endl;
 #endif
-            dpy = XOpenDisplay(NULL);
+            assert(errno == 0 && "Errno should not be set at start of xlib_gl_extended_window constructor");
+
+            dpy = XOpenDisplay(nullptr);
             if (!dpy) {
-                printf("\n\tcannot connect to X server\n\n");
-                exit(1);
+                std::cerr << "Cannot connect to X server" << std::endl;
+                std::exit(1);
             } else {
 				// Apparently, XOpenDisplay's _true_ error indication is whether dpy is nullptr.
 				// https://cboard.cprogramming.com/linux-programming/119957-xlib-perversity.html
 				// if (errno != 0) {
 				// 	std::cerr << "XOpenDisplay succeeded, but errno = " << errno << "; This is benign, so I'm clearing it now.\n";
 				// }
-				errno = 0;
+				RAC_ERRNO_MSG("extended_window after XOpenDisplay : value returned not null");
```

Comment:
The point of this is that non-zero `errno` here doesn't actually denote an erroneous condition.

Any output caused by this is noise. Let's go back to `errno = 0`.

Reply:
Agreed the output is noise, but currently, `RAC_ERRNO` and `RAC_ERRNO_MSG` only print and clear when `NDEBUG` is on. Errnos throughout execution during a run with `opt` are ignored. Clearing errno here would be the exception.

Reply:
Debug mode will be active when someone is hunting for the source of a true bug. Getting a print for a not-genuine-error in such case would be misleading. The reader has to read the message and understand the source to determine that it is not genuine.